File: src/Chapter-3-Rust/Questions/Question4.rs
//Write a program that prompts the user to enter a telephone number in
//the form (xxx) xxx-xxxx and then displays the number in the form xxx.xxx.xxx:
//Enter phone number [(xxx) xxx-xxxx]: (404) 817-6900
//You entered 404.817.6900
#[macro_use] extern crate text_io;
//to pause console
use std::process::Command;
//to use print!
use std::io::{self, Write};
fn main()
{
	let (first, second, third) : (i32, i32, i32);
	print!("Enter phone number [(xxx) xxx-xxxx]: ");
	io::stdout().flush().unwrap();
	scan!("({}) {}-{}", first, second, third);
	print!("You entered {}.{}.{}\n", first, second, third);
	io::stdout().flush().unwrap();
	let _ = Command::new("cmd.exe").arg("/c").arg("pause").status();
}

File: src/Chapter-7-C99/Questions/Question4.c
//Write a program that translates an alphabetic phone number into numeric form:
//Enter phone number: CALLATT
//2255288
#include <stdio.h>

int main(void)
{
    char ch;
    printf("Enter phone number: ");
    while((ch = getchar()) != '\n')
    {
        if (ch >= 65 && ch <= 67)
        {
            printf("2");
        }
        else if (ch >= 68 && ch <= 70)
        {
            printf("3");
        }
        else if (ch >= 71 && ch <= 73)
        {
            printf("4");
        }
        else if (ch >= 74 && ch <= 76)
        {
            printf("5");
        }
        else if (ch >= 77 && ch <= 79)
        {
            printf("6");
        }
        else if (ch >= 80 && ch <= 82)
        {
            printf("7");
        }
        else if (ch >= 83 && ch <= 85)
        {
            printf("8");
        }
        else if (ch >= 86 && ch <= 88)
        {
            printf("9");
        }
        else
        {
            printf("%c", ch);
        }

    }
    return 0;
}
